client/src/app/site/motions/components/motion-list/motion-list.component.ts

```diff
@@ -187,9 +195,11 @@ export class MotionListComponent extends ListViewBaseComponent<ViewMotion> imple
      * in the ui
      * @param status TODO: May still change type
      */
-    public setStatusSelected(status: Partial<WorkflowState>): void { // TODO: check if id is there
+    public async setStatusSelected(status: Partial<WorkflowState>): Promise<void> { // TODO: check if id is there
         this.selectedRows.forEach(vm => {
-            this.repo.update({state_id: status.id}, vm);
+            this.repo.update({state_id: status.id}, vm).then(f => {
```

Comment:
the approach that was used most times was:
`.then(null, this.raiseError);`

Reply:
.. which you won't want in multiSelect. Or you'll have up to 400 'OK!' popping up very fast.
This is basically the 'next iteration, please' of async inside forEach.
As long as there are no real errors coming, it makes no sense to insert a counter here ('3 deletions failed').

And I'm not very happy with the pseudo-asynchronousity of repository actions (having to wait for one action to end so that the next may proceed). Is this really necessary?

Reply:
in this case, you might just add
`await this.repo.update({state_id: status.id}, vm);`
which would hopefully just execute them one after another.

An alternative approach would be to store the promises and then execute them with promise.all()

If you want to fake the .then, the syntax we used to use was: `.then(() => {})`

Reply:
Changed it into firing all requests and returning a reaction after Promise.all has been completed. We'll have to see how that compares to 'I start doing $something, change the view, $something finishes'

Reply:
... and changed back, because the server currently gets a locked database and doesn't queue.